designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputDateTime.kt

```diff
@@ -438,6 +455,381 @@ fun getSupportingTextList(uiModel: InputDateTimeModel, dateOutOfRangeItem: Suppo
+        supportingTextList.add(item)
+    }
+    if (!inputTextFieldValue?.text.isNullOrEmpty()) {
+        val dateIsInRange: Boolean
+        val dateIsInYearRange: Boolean
+        val isValidHourFormat: Boolean
+        val isValidDateFormat: Boolean
+
+        when (actionType) {
+            DateTimeActionType.TIME -> {
+                if (inputTextFieldValue?.text!!.length == 4) {
+                    isValidHourFormat = isValidHourFormat(inputTextFieldValue.text)
+                    if (!isValidHourFormat) supportingTextList.add(incorrectHourFormatItem)
+                    supportingText
+                }
+            }
+            DateTimeActionType.DATE_TIME -> {
+                if (inputTextFieldValue?.text!!.length == 12) {
+                    dateIsInRange = dateIsInRange(
+                        parseStringDateToMillis(
+                            inputTextFieldValue.text.substring(0, inputTextFieldValue.text.length - 4),
+                        ),
+                        selectableDates,
+                    )
+                    dateIsInYearRange = yearIsInRange(inputTextFieldValue.text, getDefaultFormat(actionType), yearRange)
+                    isValidHourFormat = isValidHourFormat(inputTextFieldValue.text.substring(8, 12))
+                    isValidDateFormat = isValidDate(inputTextFieldValue.text.substring(0, 8))
+                    if (!dateIsInRange || !dateIsInYearRange) supportingTextList.add(dateOutOfRangeItem)
+                    if (!isValidDateFormat) supportingTextList.add(incorrectDateFormatItem)
+                    if (!isValidHourFormat) supportingTextList.add(incorrectHourFormatItem)
+                }
+            }
+            DateTimeActionType.DATE -> {
+                if (inputTextFieldValue?.text!!.length == 8) {
+                    dateIsInRange = dateIsInRange(parseStringDateToMillis(inputTextFieldValue.text), selectableDates)
+                    isValidDateFormat = isValidDate(inputTextFieldValue.text)
+                    dateIsInYearRange = yearIsInRange(inputTextFieldValue.text, getDefaultFormat(actionType), yearRange)
+                    if (!dateIsInRange || !dateIsInYearRange) supportingTextList.add(dateOutOfRangeItem)
+                    if (!isValidDateFormat) supportingTextList.add(incorrectDateFormatItem)
+                }
+            }
+        }
+    }
+    return supportingTextList.toList()
+}
+
+/**
+ * DHIS2 Input Date Time
+ * Input field to enter date, time or date&time. It will format content based on given visual
+ * transformation.
+ * component uses Material 3 [DatePicker] and [TimePicker]
+ * input formats supported are mentioned in the date time input ui model documentation.
+ * [DatePicker] Input mode  will always follow locale format.
+ * @param uiModel: an [InputDateTimeModel] with all the parameters for the input
+ * @param modifier: optional modifier.
+ */
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun InputDateTime(
```

Comment:
Can you create unit test to cover some examples of dates?

Reply:
will do!